infra/event-sync/src/main/java/com/simprints/infra/eventsync/event/remote/models/session/ApiEventScopeEndCause.kt

```diff
@@ -0,0 +1,16 @@
+package com.simprints.infra.eventsync.event.remote.models.session
+
+import androidx.annotation.Keep
+import com.simprints.infra.events.event.domain.models.scope.EventScopeEndCause
+
+@Keep
+internal enum class ApiEventScopeEndCause {
+
+    WORKFLOW_ENDED,
+    NEW_SESSION,
+}
+
+internal fun EventScopeEndCause?.fromDomainToApi(): ApiEventScopeEndCause = when (this) {
+    null, EventScopeEndCause.WORKFLOW_ENDED -> ApiEventScopeEndCause.WORKFLOW_ENDED
```

Comment:
From the business logic, is it correct to associate `null` with a finished workflow?

Reply:
IMO, it is ok in this case since this is the default fallback value. The only other end cause is NEW_SESSION which is the one we actually care about.